x/evm/types/turnstone_abi.go

```diff
@@ -98,19 +141,45 @@ func (_m *Message_SubmitLogicCall) keccak256(orig *Message, nonce uint64) []byte
 			{Name: "address", Type: "address"},
 			{Name: "payload", Type: "bytes"},
 		}))},
+		// fee arguments
+		{Type: whoops.Must(abi.NewType("tuple", "", []abi.ArgumentMarshaling{
+			{Name: "relayer_fee", Type: "uint256"},
+			{Name: "community_fee", Type: "uint256"},
+			{Name: "security_fee", Type: "uint256"},
+			{Name: "fee_payer_paloma_address", Type: "bytes32"},
+		}))},
 		// message id
 		{Type: whoops.Must(abi.NewType("uint256", "", nil))},
 		// turnstone id
 		{Type: whoops.Must(abi.NewType("bytes32", "", nil))},
 		// deadline
 		{Type: whoops.Must(abi.NewType("uint256", "", nil))},
+		// relayer
+		{Type: whoops.Must(abi.NewType("address", "", nil))},
 	}
 
 	method := abi.NewMethod("logic_call", "logic_call", abi.Function, "", false, false, arguments, abi.Arguments{})
 	var bytes32 [32]byte
 
 	copy(bytes32[:], orig.GetTurnstoneID())
 
+	var fees *SubmitLogicCall_Fees
+	if m.Fees == nil {
+		// If we have no fees set in the message, we use the same defaults as
+		// pigeon
+		fees = &SubmitLogicCall_Fees{
+			RelayerFee:   100_000,
+			CommunityFee: 100_000,
+			SecurityFee:  100_000,
+		}
+	} else {
+		fees = m.Fees
+	}
+
+	// Left-pad the address with zeroes
+	padding := bytes.Repeat([]byte{0}, 32-len(m.SenderAddress))
+	senderAddress := [32]byte(append(padding, m.SenderAddress...))
```

Comment:
Pigeon already performs this bit: https://github.com/palomachain/pigeon/blob/02c99aea1e69913a6b026a8cbeef718cf7bad4f2/chain/evm/compass.go#L257

But I guess it won't hurt.

Reply:
We need to do the same as pigeon here, so the hashes match, right?

Reply:
Yes, yes we do ...